pkg/cli/sync.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/abcxyz/pkg/cli"
+)
+
+var (
+	_                        cli.Command = (*SyncCommand)(nil)
+	allowedSourceSystem                  = []string{"GOOGLEGROUP"}
+	allowedDestinationSystem             = []string{"GITHUB"}
+)
+
+type SyncCommand struct {
+	cli.BaseCommand
+
+	source             string
+	destination        string
+	groupMappingConfig string
+	userMappingConfig  string
+	sourceToken        string
+	destinationToken   string
+}
+
+func (c *SyncCommand) Desc() string {
+	return `Sync membership information`
+}
+
+func (c *SyncCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+  Sync membership information from source and target system.
+
+  Sync membership from google group to GitHub
+
+  tlctl sync run \
+	-src GoogleGroup \
+	-dest GitHub \
+	-group-mapping-config your-group-config.textproto \
+	-user-mapping-config your-user-confif.textproto
+`
+}
+
+func (c *SyncCommand) Flags() *cli.FlagSet {
+	set := c.NewFlagSet()
+
+	// Command options
+	f := set.NewSection("COMMAND OPTIONS")
+
+	f.StringVar(&cli.StringVar{
+		Name:    "source",
+		Target:  &c.source,
+		Aliases: []string{"src", "s"},
+		Example: "GoogleGroup",
+		Usage:   `The source system to read membership `,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "destination",
+		Target:  &c.destination,
+		Aliases: []string{"d", "dst"},
+		Example: "GitHub",
+		Usage:   `The target system for membership information`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "group-mapping-config",
+		Target:  &c.groupMappingConfig,
+		Aliases: []string{"g", "gc", "group-config", "g-config"},
```

Comment:
Why so many aliases?

Reply:
I was thinking to provide users more choices, now I think it's unnecessary. Removed, and only kept `gc` and `uc`